web/thesauruses/kotlin/1.5/language_basics.json

```diff
@@ -0,0 +1,97 @@
+{
+  "meta": {
+    "language": "kotlin",
+    "language_version": "1.5",
+    "language_name": "Kotlin",
+    "structure": "language_basics"
+  },
+  "concepts": {
+    "statically_typed": {
+      "code": "val x: Int = 5"
+    },
+    "dynamically_typed": {
+      "not-implemented": true
+    },
+    "procedural_language": {
+      "code": "fun calculateSum(a: Int, b: Int): Int {\n    return a + b\n}"
+    },
+    "functional_language": {
+      "code": "val numbers = listOf(1, 2, 3, 4, 5)\nval doubled = numbers.map { it * 2 }"
+    },
+    "object_oriented_language": {
+      "code": "class Person(val name: String, val age: Int)"
+    },
+    "low_level_language": {
+      "not-implemented": true
+    },
+    "high_level_language": {
+      "code": "println(\"Hello, World!\")"
+    },
+    "compiled_language": {
+      "code": "// Kotlin compiles to JVM bytecode or native code",
+      "comment": "Kotlin is compiled to JVM bytecode, JavaScript, or native binaries"
+    },
+    "interpreted_language": {
+      "not-implemented": true
+    },
+    "general_purpose_language": {
+      "code": "// Kotlin can be used for Android, web, server-side, desktop, etc."
+    },
+    "domain_specific_language": {
+      "not-implemented": true
+    },
+    "manual_management": {
+      "not-implemented": true
+    },
+    "automatic_management": {
+      "code": "// Kotlin uses JVM's garbage collector",
+      "comment": "Memory is managed automatically by the JVM garbage collector"
+    },
+    "first_generation": {
+      "not-implemented": true
+    },
+    "second_generation": {
+      "not-implemented": true
+    },
+    "third_generation": {
+      "code": "// Kotlin is a modern third-generation language"
+    },
+    "fourth_generation": {
+      "not-implemented": true
+    },
```

Comment:
For this template, the intention is that these are more true/false types of things if the language supports it. So they would all have code blocks that specify true/false.  (I realize no language has implemented this yet so you may not have known. The discussion behind is in #679 ).